### PR TITLE
feat: support unquoted HTML attributes, closes #963

### LINKED ISSUE #963: templ errors if HTML attributes aren't quoted
**Describe the bug**

If I try run `templ` on this file:

```
package main

templ Hello() {
  <div id=main></div>
}
```

I get this error:

```
parsing error: <div>: malformed open element: line 3, col 7
```

Seems related to #116 / #125.

**Expected behavior**

`<div id=main></div>` is valid HTML ([this post discusses how unquoted attributed values are valid](https://mathiasbynens.be/notes/unquoted-attribute-values)), so I expected `templ` to compile it even if it's a little unusual.

That said I noticed that the [Elements documentation page](https://templ.guide/syntax-and-usage/elements) documents ways in which templ diverges from HTML ("Unlike HTML, templ requires that all HTML elements are closed..."), so if this is intentional it might be good to document it.

**`templ info` output**

```
(✓) os [ goos=darwin goarch=arm64 ]
(✓) go [ location=/opt/homebrew/bin/go version=go version go1.23.1 darwin/arm64 ]
(✓) gopls [ location=/Users/bork/go/bin/gopls version=golang.org/x/tools/gopls v0.8.4
    golang.org/x/tools/gopls@v0.8.4 h1:zGZsAXAb0LLws/Z+2BCWR17dkPHhIO2GYwYSSkeXX5c= ]
(✓) templ [ location=/Users/bork/go/bin/templ version=v0.2.778 ]
```

**Desktop (please complete the following information):**

 - OS: Mac OS
 - templ CLI version? v0.2.778
- Go version: 1.23
- `gopls` version: v0.8.4

### PR DESCRIPTION
**Pull request by a-h**
- **feat: support unquoted attribute values**
- **feat: add tests for unquoted attributes**

Closes #963

### CODE REVIEW

File: parser/v2/elementparser.go
Comment (joerdav) [inline parser/v2/elementparser.go:105]:
```suggestion
	// A valid unquoted attribute value in HTML is any string of text that is not an empty string and that doesn’t contain spaces, tabs, line feeds, form feeds, carriage returns, ", ', `, =, <, or >.
```

File: parser/v2/elementparser.go
Comment (joerdav) [inline parser/v2/elementparser.go:124]: Nit - these can be converted to arrays `[...]` or defined outside of this function to avoid some allocations.

**Review by joerdav — CHANGES_REQUESTED** [review]
Looks all good, initially the fact that the values aren't quoted worried me a bit that we may get a greedy parser here but I don't think that'll be a problem, the end condition is fairly straight forward. Had some nit picks that I'm happy to approve to go in if you are?

I also thought a fuzz test could be good, but that may be overkill here.

File: parser/v2/elementparser.go
Comment (a-h) [reply parser/v2/elementparser.go:124]: I thought the same thing. I'll do a refactor.

**Comment by a-h** [comment]
OK, this is ready for a second review. Thanks for that!

**Review by joerdav — APPROVED** [review]
LGTM

**Pull request merged**
